pkg/cacheutil/memcached_client.go

```diff
@@ -462,11 +462,10 @@ func (c *memcachedClient) resolveAddrs() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 
-	c.dnsProvider.Resolve(ctx, c.config.Addresses)
-
+	err := c.dnsProvider.Resolve(ctx, c.config.Addresses)
 	// Fail in case no server address is resolved.
 	servers := c.dnsProvider.Addresses()
-	if len(servers) == 0 {
+	if len(servers) == 0 || err != nil {
```

Comment:
I think we want to always check error, no?

Reply:
Sure, made a note.